intake_xarray/image.py

```diff
@@ -111,7 +113,54 @@ def add_leading_dimension(x):
     return Array(dsk, name, chunks, sample.dtype)
 
 
-def reader(file, chunks, imread=None, preprocess=None, coerce_shape=None):
+def _dask_exifread(files, exif_tags):
+    """Construct a dask Array to read each tag in `exif_tags` (list of
+    str) from the EXIF data of the images in `files`
+    """
+    from numpy import array
+    from dask.array import Array
+    from dask.base import tokenize
+    from exifread import process_file as read_exif
+
+    def _read_exif(open_file):
+        # Using the context manager (as below) occasionally results
+        # in 'I/O operation on closed file' and similar errors
+        # with open_file as f:
+        #     return read_exif(f)
+        #
+        f = open_file.open()
+        return read_exif(f)
```

Comment:
I'm not sure of the root cause of this problem (or how robust the workaround is).

I did not see this behaviour when using `with open(open_file.path, 'rb') as f:` instead of the fsspec context manager.

Reply:
It is not a problem to leave the file "open", it doesn't hold a lot of system resources except for some memory buffer, which will be released when the reader is released.

Reply:
I've since encountered 'too many open files' errors when not explicitly closing them, so perhaps worth investigating. Any idea why the context manager might be failing?

Reply:
I suppose the reader keeps a reference to the open file?
Basically, you want to make sure that each dask task opens a file from scratch, and returns an in-memory array (and not any references to the reader/file).

Reply:
I don't think so, unless `read_exif` is doing something unexpected. I checked with an alternative that definitely didn't keep hold of the file and still saw these errors.

I think the problem might be related to passing the same OpenFile instances both to this function and to `_dask_imread` - I don't see the error if I open the file in only one of these functions, or if I copy the OpenFile object (as in the latest commit).